manifests/init.pp

```diff
@@ -86,12 +87,17 @@
   $pool_test_on_borrow                                              = false,
   # JVM Settings
   $javahome                                                         = undef,
-  $jvm_xms                                                          = '256m',
-  $jvm_xmx                                                          = '1024m',
-  $jvm_permgen                                                      = '256m',
-  $jvm_optional                                                     = '-XX:-HeapDumpOnOutOfMemoryError',
-  $java_opts                                                        = '',
-  $catalina_opts                                                    = '',
+  Enum['openjdk-11', 'oracle-jdk-1.8', 'custom'] $jvm_type          = 'custom',
```

Comment:
given that this might just get more, it might be better to extract it in its own type